jni/src/main/java/com/horizen/certnative/NaiveThresholdSignatureWKeyRotation.java

```diff
@@ -0,0 +1,759 @@
+     * @param btrFee - fee for BackwardTransfer
+     * @param ftMinAmount - minimum amount for Forward Transfer
+     * @param constant - constant parameter, as defined by getConstant() method
+     * @param quality - quality parameter, as returned by the createProof() function (in this case the number of valid signatures)
+     * @param customFields - additional parameters. Can be empty.
+     * @param proof - the bytes of the proof to be verified, expected to be in compressed form
+     * @param checkProof - enable semantic checks on the proof
+     * @param verificationKeyPath - file path from which loading the verification key, expected to be in compressed form
+     * @param checkVerificationKey - enable semantic checks on the verification key
+     * @return true, if proof verification was successfull, false if proof verification failed or if some errors occured during verification
+     */
+    public static boolean verifyProof(
+            List<BackwardTransfer> btList,
+            FieldElement scId,
+            int epochNumber,
+            FieldElement endCumulativeScTxCommTreeRoot,
+            long btrFee,
+            long ftMinAmount,
+            FieldElement constant,
+            long quality,
+            List<FieldElement> customFields,
+            byte[] proof,
+            boolean checkProof,
+            String verificationKeyPath,
+            boolean checkVerificationKey
+    )
+    {
+        return nativeVerifyProof(
+            btList.toArray(new BackwardTransfer[0]), scId, epochNumber,
+            endCumulativeScTxCommTreeRoot, btrFee, ftMinAmount,
+            constant, quality, customFields.toArray(new FieldElement[0]),
+            proof, checkProof, true, verificationKeyPath, checkVerificationKey, true
+        );
+    }
+
+    /**
+     * Verify proof using the supplied parameters
+     * @param btList - the list of backward transfer for a given certificate
+     * @param scId - the id of the corresponding sidechain
+     * @param epochNumber - the epoch number for the certificate
+     * @param endCumulativeScTxCommTreeRoot - the value of the cumulative sidechain transaction commitment tree at epoch end
+     * @param btrFee - fee for BackwardTransfer
+     * @param ftMinAmount - minimum amount for Forward Transfer
+     * @param constant - constant parameter, as defined by getConstant() method
+     * @param quality - quality parameter, as returned by the createProof() function (in this case the number of valid signatures)
+     * @param customFields - additional parameters. Can be empty.
+     * @param proof - the bytes of the proof to be verified, expected to be in compressed form
+     * @param verificationKeyPath - file path from which loading the verification key, expected to be in compressed form
+     * @return true, if proof verification was successfull, false if proof verification failed or if some errors occured during verification
+     */
+    public static boolean verifyProof(
```

Comment:
We also need the sc_prev_wcert_hash in order to verify the proof, I don't see it here